modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIPropertyRelevantObjectEntryModelListener.java

```diff
@@ -43,15 +46,54 @@ public String getObjectDefinitionExternalReferenceCode() {
 	public void onBeforeCreate(ObjectEntry objectEntry)
 		throws ModelListenerException {
 
-		_validate(objectEntry);
+		_validate(objectEntry.getValues());
 	}
 
 	@Override
 	public void onBeforeUpdate(
 			ObjectEntry originalObjectEntry, ObjectEntry objectEntry)
 		throws ModelListenerException {
 
-		_validate(objectEntry);
+		Map<String, Serializable> values = objectEntry.getValues();
+
+		long apiSchemaId = (long)values.get(
+			"r_apiSchemaToAPIProperties_c_apiSchemaId");
+
+		if (apiSchemaId != 0) {
+			_validate(values);
+		}
+
+		_scheduleAPIProperties(objectEntry.getObjectEntryId());
+	}
+
+	private void _scheduleAPIProperties(long apiPropertyId) {
```

Comment:
Maybe another name would help to understand why we are doing this, such as:

`_scheduleOrphanAPIPropertyDeletion`